src/Gpg/ImportGpgKeyFromStringViaTemporaryFile.php

```diff
@@ -4,35 +4,27 @@
 
 namespace Laminas\AutomaticReleases\Gpg;
 
-use Symfony\Component\Process\Process;
-use Webmozart\Assert\Assert;
-
-use function Safe\file_put_contents;
-use function Safe\preg_match;
-use function Safe\tempnam;
-use function sys_get_temp_dir;
-use function unlink;
+use Psl;
+use Psl\Env;
+use Psl\Filesystem;
+use Psl\Regex;
+use Psl\Shell;
 
 final class ImportGpgKeyFromStringViaTemporaryFile implements ImportGpgKeyFromString
 {
     public function __invoke(string $keyContents): SecretKeyId
     {
-        $keyFileName = tempnam(sys_get_temp_dir(), 'imported-key');
-
-        file_put_contents($keyFileName, $keyContents);
-
-        $output = (new Process(['gpg', '--import', $keyFileName]))
-            ->mustRun()
-            ->getErrorOutput();
+        $keyFileName = Filesystem\create_temporary_file(Env\temp_dir(), 'imported-key');
+        Filesystem\write_file($keyFileName, $keyContents);
 
-        Assert::regex($output, '/key\\s+([A-F0-9]+):\\s+secret\\s+key\\s+imported/im');
+        // redirect output from STDERR to STDOUT since Shell\execute only returns STDOUT content.
```

Comment:
Is there a test for this redirect? Escaping `2>&1` is kinda fiddly

Reply:
not sure how to test it? escaping is disabled here.

Reply:
Possibly not testable: just saw that STDERR has to be piped at process open, if you want to capture it.

Looking at https://github.com/laminas/automatic-releases/pull/138/checks?check_run_id=2638984783#step:8:11, I see no STDERR, so I suppose this doesn't need testing here.

This still brings me to another point: symfony CLI does escape shell arguments by default, what happens with this library instead? Is this the last argument of `Shell\execute()`?

Reply:
yes, Shell\execute also escapes by default, unless the last argument is set to false ( which is done here, and we escape the only argument that needs escaping ).

see https://github.com/azjezz/psl/blob/1.8.x/src/Psl/Shell/execute.php#L44

Reply:
Excellent, thanks!